engine/frontend/src/component/PackageCatalogForm.js

```diff
@@ -101,6 +101,20 @@ const checkValidIntType = (data) => {
     }
 }
 
+const checkValidFloatType = (data) => {
+    const  isValidFloat = (value) => {
+        return (/^-?[\d]*(\.[\d]+)?$/g).test(value);
```

Comment:
@Peeeekay It's not wrong to use a regex per se (I haven't tested the validity of this), but can we not use Number.parsefloat or [what this guy is saying](https://stackoverflow.com/a/52740766)?

Reply:
I have used the regex that was was the second answer in the slack link. IsNaN has an edge-case where "" returns true; for our initial usecases the regex seems appealing. Parsefloat also has issues where it returns true for some of failure modes. @adschwartz

Reply:
I think we can just check for an empty string no? i.e. if you get `""` then it's not a float.

Reply:
hm ok; i can do that.

Reply:
oh nevermind; i was looking at different link.